pkgs/applications/office/docear/default.nix

```diff
@@ -0,0 +1,48 @@
+{stdenv, fetchurl, runtimeShell
+, oraclejre
+, antialiasFont ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "docear";
+  version = "1.2";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "http://docear.org/downloads/docear_linux.tar.gz";
+    sha256 = "1g5n7r2x4gas6dl2fbyh7v9yxdcb6bzml8n3ldmpzv1rncgjcdp4";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+  buildInputs = [ oraclejre ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share
+    cp -R * $out/share
+    chmod 0755 $out/share/ -R
+
+    # The wrapper ensures oraclejre is used
+    echo "#!${runtimeShell}" > $out/bin/docear
```

Comment:
You're looking for `makeWrapper` here. If you do a search on github for `makeWrapper` + `jre` I'm sure you'll find a number of examples. If not, feel free to ask for further help.